internal/rbd/types/snapshot.go

```diff
@@ -37,4 +40,16 @@ type Snapshot interface {
 
 	// SetVolumeGroup sets the CSI volume group ID in the snapshot.
 	SetVolumeGroup(ctx context.Context, creds *util.Credentials, vgID string) error
+
+	// GetSize returns the size of the snapshot in bytes.
+	GetSize() int64
+
+	// ProcessMetadata processes the block metadata for a snapshot.
+	// If baseRBDSnap is provided, the delta between the
+	// current snapshot and the base snapshot is processed.
+	ProcessMetadata(ctx context.Context,
```

Comment:
a common coding practice is to have the callback as last parameter for functions (with optional callback arguments after that)